src/main/java/uk/gov/pay/adminusers/service/InviteService.java

```diff
@@ -131,7 +134,21 @@ public void generateOtp(InviteOtpRequest inviteOtpRequest) {
                     });
             LOGGER.info("New 2FA token generated for invite code [{}]", inviteOtpRequest.getCode());
         } else {
-            throw notFoundException();
+            throw notFoundInviteException(inviteOtpRequest.getCode());
+        }
+    }
+
+    @Transactional
+    public User createInvitedUser(InviteValidateOtpRequest inviteValidateOtpRequest) {
+        Optional<InviteEntity> inviteOptional = inviteDao.findByCode(inviteValidateOtpRequest.getCode());
+        if (inviteOptional.isPresent() && secondFactorAuthenticator.authorize(inviteOptional.get().getOtpKey(), inviteValidateOtpRequest.getOtpCode())) {
```

Comment:
I have added `401 - UNAUTHORIZED` with the log message `Invite for code %s provided invalid otp auth code`